File: frontend/api_client.py
import os
import httpx
from typing import Optional, Dict, Any

API_BASE_URL = os.getenv("API_BASE_URL", "http://localhost:8000")


class APIClient:
    def __init__(self, token: Optional[str] = None):
        self.base_url = f"{API_BASE_URL}/api/v1"
        self.token = token

    def _headers(self) -> Dict[str, str]:
        headers = {"Content-Type": "application/json"}
        if self.token:
            headers["Authorization"] = f"Bearer {self.token}"
        return headers

    async def _request(
        self,
        method: str,
        endpoint: str,
        data: Optional[Dict] = None,
        params: Optional[Dict] = None,
    ) -> Dict[str, Any]:
        async with httpx.AsyncClient() as client:
            response = await client.request(
                method,
                f"{self.base_url}{endpoint}",
                json=data,
                params=params,
                headers=self._headers(),
                timeout=30.0,
            )
            if response.status_code >= 400:
                error = response.json()
                raise APIError(response.status_code, error)
            if response.status_code == 204:
                return {}
            return response.json()

    def _request_sync(
        self,
        method: str,
        endpoint: str,
        data: Optional[Dict] = None,
        params: Optional[Dict] = None,
    ) -> Dict[str, Any]:
        try:
            with httpx.Client() as client:
                response = client.request(
                    method,
                    f"{self.base_url}{endpoint}",
                    json=data,
                    params=params,
                    headers=self._headers(),
                    timeout=30.0,
                )
                if response.status_code >= 400:
                    try:
                        error = response.json()
                    except:
                        error = {"detail": {"code": "API_ERROR", "message": f"HTTP {response.status_code}: {response.text or 'No response'}"}}
                    raise APIError(response.status_code, error)
                if response.status_code == 204:
                    return {}
                return response.json()
        except httpx.ConnectError:
            raise APIError(503, {"detail": {"code": "CONNECTION_ERROR", "message": "Cannot connect to API server. Is the backend running?"}})

    # Auth
    def signup(self, email: str, password: str, role: str, display_name: str = None, business_name: str = None):
        data = {"email": email, "password": password, "role": role}
        if display_name:
            data["display_name"] = display_name
        if business_name:
            data["business_name"] = business_name
        return self._request_sync("POST", "/auth/signup", data)

    def login(self, email: str, password: str):
        return self._request_sync("POST", "/auth/login", {"email": email, "password": password})

    def get_me(self):
        return self._request_sync("GET", "/auth/me")

    # Profiles
    def get_my_instructor_profile(self):
        return self._request_sync("GET", "/instructors/me")

    def update_instructor_profile(self, data: Dict):
        return self._request_sync("PUT", "/instructors/me", data)

    def get_my_studio_profile(self):
        return self._request_sync("GET", "/studios/me")

    def update_studio_profile(self, data: Dict):
        return self._request_sync("PUT", "/studios/me", data)

    def get_instructor(self, instructor_id: str):
        return self._request_sync("GET", f"/instructors/{instructor_id}")

    def get_studio(self, studio_id: str):
        return self._request_sync("GET", f"/studios/{studio_id}")

    # Job Posts
    def create_job_post(self, data: Dict):
        return self._request_sync("POST", "/job-posts", data)

    def list_job_posts(self, params: Optional[Dict] = None):
        return self._request_sync("GET", "/job-posts", params=params)

    def list_job_posts_with_matching(self, params: Optional[Dict] = None):
        """List job posts with matching scores for instructors."""
        return self._request_sync("GET", "/job-posts/for-me/with-matching", params=params)

    def get_job_post(self, job_post_id: str):
        return self._request_sync("GET", f"/job-posts/{job_post_id}")

    def update_job_post(self, job_post_id: str, data: Dict):
        return self._request_sync("PUT", f"/job-posts/{job_post_id}", data)

    def delete_job_post(self, job_post_id: str):
        return self._request_sync("DELETE", f"/job-posts/{job_post_id}")

    # Applications
    def apply_to_job(self, job_post_id: str, cover_letter: str = None):
        data = {}
        if cover_letter:
            data["cover_letter"] = cover_letter
        return self._request_sync("POST", f"/job-posts/{job_post_id}/applications", data)

    def get_my_applications(self):
        return self._request_sync("GET", "/applications/me")

    def withdraw_application(self, application_id: str):
        return self._request_sync("POST", f"/applications/{application_id}/withdraw")

    def get_job_post_applications(self, job_post_id: str):
        """Get all applications for a job post (studio owner only)."""
        return self._request_sync("GET", f"/job-posts/{job_post_id}/applications")

    # Offers
    def create_offer(self, data: Dict):
        return self._request_sync("POST", "/offers", data)

    def get_my_offers(self):
        return self._request_sync("GET", "/offers/me")

    def accept_offer(self, offer_id: str):
        return self._request_sync("POST", f"/offers/{offer_id}/accept")

    def reject_offer(self, offer_id: str):
        return self._request_sync("POST", f"/offers/{offer_id}/reject")

    # Contracts
    def create_contract_from_offer(self, offer_id: str):
        return self._request_sync("POST", f"/contracts/from-offer/{offer_id}")

    def get_my_contracts(self):
        return self._request_sync("GET", "/contracts/me")

    def set_contract_in_progress(self, contract_id: str):
        return self._request_sync("POST", f"/contracts/{contract_id}/set-in-progress")

    def complete_contract(self, contract_id: str):
        """Confirm contract completion (v2.0 - bidirectional confirmation)."""
        return self._request_sync("POST", f"/contracts/{contract_id}/confirm-completion")

    def cancel_contract(self, contract_id: str, reason: str):
        return self._request_sync("POST", f"/contracts/{contract_id}/cancel", {"reason": reason})

    def report_no_show(self, contract_id: str, reported_user_id: str):
        """Report a no-show for a contract."""
        return self._request_sync("POST", f"/contracts/{contract_id}/report-no-show", {
            "reported_user_id": reported_user_id
        })

    # Payments
    def initialize_payment(self, contract_id: str):
        return self._request_sync("POST", f"/contracts/{contract_id}/payments")

    def confirm_payment(self, payment_key: str, order_id: str, amount: float):
        return self._request_sync("POST", "/payments/confirm", {
            "payment_key": payment_key,
            "order_id": order_id,
            "amount": amount,
        })

    # Chat
    def get_threads(self):
        return self._request_sync("GET", "/threads")

    def create_thread(self, data: Dict):
        return self._request_sync("POST", "/threads", data)

    def get_thread_messages(self, thread_id: str):
        return self._request_sync("GET", f"/threads/{thread_id}/messages")

    def send_message(self, thread_id: str, content: str):
        return self._request_sync("POST", f"/threads/{thread_id}/messages", {"content": content})

    # Reviews
    def create_review(self, contract_id: str, rating: int, comment: str = None):
        data = {"rating": rating}
        if comment:
            data["comment"] = comment
        return self._request_sync("POST", f"/contracts/{contract_id}/reviews", data)

    def get_my_review_for_contract(self, contract_id: str):
        """Get current user's review for a specific contract."""
        return self._request_sync("GET", f"/contracts/{contract_id}/reviews/my")

    def update_review(self, review_id: str, rating: int = None, comment: str = None):
        """Update an existing review."""
        data = {}
        if rating is not None:
            data["rating"] = rating
        if comment is not None:
            data["comment"] = comment
        return self._request_sync("PUT", f"/reviews/{review_id}", data)

    def delete_review(self, review_id: str):
        """Delete a review."""
        return self._request_sync("DELETE", f"/reviews/{review_id}")

    # Support
    def create_support_ticket(self, subject: str, description: str):
        return self._request_sync("POST", "/support/tickets", {
            "subject": subject,
            "description": description,
        })

    def get_my_support_tickets(self):
        return self._request_sync("GET", "/support/tickets/me")

    # Verification
    def request_phone_verification(self, phone: str):
        return self._request_sync("POST", "/verification/phone/request", {"phone": phone})

    def verify_phone(self, phone: str, otp: str):
        return self._request_sync("POST", "/verification/phone/verify", {"phone": phone, "otp": otp})

    def verify_business(self, business_number: str):
        return self._request_sync("POST", "/verification/business/verify", {"business_number": business_number})

    def get_verification_status(self):
        return self._request_sync("GET", "/verification/status")

    # Deposit
    def get_deposit_status(self):
        return self._request_sync("GET", "/deposit/status")

    def add_deposit(self, amount: float):
        return self._request_sync("POST", "/deposit/add", {"amount": amount})

    # Subscription (Premium Membership)
    def get_subscription_status(self):
        """Get current user's subscription status."""
        return self._request_sync("GET", "/subscriptions/me")

    def initialize_premium_upgrade(self):
        """Initialize premium subscription upgrade."""
        return self._request_sync("POST", "/subscriptions/upgrade", {})

    def confirm_subscription_payment(self, payment_key: str, order_id: str):
        """Confirm subscription payment after TossPayments."""
        return self._request_sync("POST", "/subscriptions/confirm", {
            "payment_key": payment_key,
            "order_id": order_id,
        })

    def cancel_subscription(self, reason: str = None):
        """Cancel active premium subscription."""
        data = {}
        if reason:
            data["reason"] = reason
        return self._request_sync("POST", "/subscriptions/cancel", data)

    def get_subscription_history(self):
        """Get subscription change history."""
        return self._request_sync("GET", "/subscriptions/history")


class APIError(Exception):
    def __init__(self, status_code: int, detail: Dict):
        self.status_code = status_code
        self.detail = detail

        # Handle different error formats
        detail_content = detail.get("detail")
        if isinstance(detail_content, list):
            # FastAPI validation error format
            errors = [f"{e.get('loc', ['?'])[-1]}: {e.get('msg', 'error')}" for e in detail_content]
            self.code = "VALIDATION_ERROR"
            self.message = "; ".join(errors)
        elif isinstance(detail_content, dict):
            # Custom error format
            self.code = detail_content.get("code", "UNKNOWN_ERROR")
            self.message = detail_content.get("message", str(detail))
        elif isinstance(detail_content, str):
            self.code = "ERROR"
            self.message = detail_content
        else:
            self.code = "UNKNOWN_ERROR"
            self.message = str(detail)

        super().__init__(self.message)

File: frontend-next/src/lib/api-client.ts
import type {
  SignupRequest,
  TokenResponse,
  MeResponse,
  InstructorProfileResponse,
  InstructorProfileUpdate,
  StudioProfileResponse,
  StudioProfileUpdate,
  JobPostCreate,
  JobPostResponse,
  JobPostListResponse,
  JobPostWithMatchingListResponse,
  ApplicationCreate,
  ApplicationResponse,
  ApplicationWithJobResponse,
  ApplicationWithInstructorResponse,
  ContactRevealResponse,
  OfferCreate,
  OfferResponse,
  OfferListResponse,
  ContractResponse,
  ContractListResponse,
  ContractCancelRequest,
  ReviewCreate,
  ReviewResponse,
  ReviewListResponse,
  ReviewEligibility,
  SubscriptionStatusResponse,
  UpgradeInitializeResponse,
  BillingKeyRegisterResponse,
  BillingMethodResponse,
  BankTransferUpgradeResponse,
  TierResponse,
  TierPublicResponse,
  TierRequirementsResponse,
  PenaltyReportRequest,
  PenaltyRecordResponse,
  MarkPaidRequest,
  PaymentConfirmationResponse,
  VerificationStatusResponse,
  ProfileCompletenessResponse,
  CertUploadResponse,
  // PMF pivot: Chat disabled -- contact reveal replaces in-app chat
  // ThreadCreate,
  // ThreadResponse,
  // MessageResponse,
  SupportTicketCreate,
  SupportTicketResponse,
  // PMF pivot: Application templates disabled
  // ApplicationTemplateResponse,
  // Daily usage removed with Trust Tier pivot
  // DailyUsageResponse,
  HandoffNoteCreate,
  HandoffNotePublicResponse,
  HandoffNoteFullResponse,
  BackupInstructorCreate,
  BackupInstructorUpdate,
  BackupInstructorResponse,
  APIErrorResponse,
} from './api-types';
import { isNativePlatform, getAccessToken } from './token-manager';

export class APIError extends Error {
  status: number;
  code: string;

  constructor(status: number, code: string, message: string) {
    super(message);
    this.name = 'APIError';
    this.status = status;
    this.code = code;
  }
}

const BASE_URL = '/api/v1';
const MAX_RETRIES = 3;
const RETRY_DELAYS = [1000, 2000];

async function request<T>(
  path: string,
  options: RequestInit = {},
): Promise<T> {
  const native = isNativePlatform();
  const baseUrl = native
    ? (process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000') + '/api/v1'
    : BASE_URL;
  const url = `${baseUrl}${path}`;

  const defaultHeaders: Record<string, string> = {};
  if (options.body && typeof options.body === 'string') {
    defaultHeaders['Content-Type'] = 'application/json';
  }

  // Native app: add Bearer token; Web: use cookies
  if (native) {
    const token = await getAccessToken();
    if (token) {
      defaultHeaders['Authorization'] = `Bearer ${token}`;
    }
  }

  const config: RequestInit = {
    ...options,
    ...(native ? {} : { credentials: 'include' as RequestCredentials }),
    headers: {
      ...defaultHeaders,
      ...options.headers,
    },
  };

  let lastError: Error | null = null;

  for (let attempt = 0; attempt < MAX_RETRIES; attempt++) {
    try {
      const response = await fetch(url, config);

      if (!response.ok) {
        const body = await response.json().catch(() => ({}));
        const detail = body.detail as APIErrorResponse | string | undefined;
        const code = typeof detail === 'object' ? detail.code : 'UNKNOWN';
        const message = typeof detail === 'object'
          ? detail.message
          : typeof detail === 'string'
            ? detail
            : `HTTP ${response.status}`;
        throw new APIError(response.status, code, message);
      }

      // 204 No Content
      if (response.status === 204) {
        return undefined as T;
      }

      return await response.json() as T;
    } catch (error) {
      lastError = error as Error;

      // 401 Unauthorized → session expired, redirect to login
      if (error instanceof APIError && error.status === 401) {
        if (typeof window !== 'undefined' && !window.location.pathname.startsWith('/login')) {
          window.location.href = `/login?callbackUrl=${encodeURIComponent(window.location.pathname)}`;
        }
        throw error;
      }

      // Don't retry client errors (4xx)
      if (error instanceof APIError && error.status >= 400 && error.status < 500) {
        throw error;
      }

      // Wait before retry (except last attempt)
      if (attempt < MAX_RETRIES - 1) {
        await new Promise((resolve) => setTimeout(resolve, RETRY_DELAYS[attempt] || 2000));
      }
    }
  }

  throw lastError || new Error('Request failed');
}

function get<T>(path: string) {
  return request<T>(path, { method: 'GET' });
}

function post<T>(path: string, body?: unknown) {
  return request<T>(path, {
    method: 'POST',
    body: body !== undefined ? JSON.stringify(body) : undefined,
  });
}

function put<T>(path: string, body?: unknown) {
  return request<T>(path, {
    method: 'PUT',
    body: body !== undefined ? JSON.stringify(body) : undefined,
  });
}

function del<T>(path: string) {
  return request<T>(path, { method: 'DELETE' });
}

function patch<T>(path: string, body?: unknown) {
  return request<T>(path, {
    method: 'PATCH',
    body: body !== undefined ? JSON.stringify(body) : undefined,
  });
}

// --- Auth ------------------------------------------------

export const auth = {
  /** Sign up (called via BFF route, not directly) */
  signup: (data: SignupRequest) =>
    post<TokenResponse>('/auth/signup', data),

  /** Login (called via BFF route, not directly) */
  login: (data: { email: string; password: string }) =>
    post<TokenResponse>('/auth/login', data),

  /** Get current user */
  me: () => get<MeResponse>('/auth/me'),
};

// --- Profiles --------------------------------------------

export const instructors = {
  getMyProfile: () =>
    get<InstructorProfileResponse>('/instructors/me'),

  updateMyProfile: (data: InstructorProfileUpdate) =>
    put<InstructorProfileResponse>('/instructors/me', data),

  getProfile: (id: string) =>
    get<InstructorProfileResponse>(`/instructors/${id}`),
};

export const studios = {
  getMyProfile: () =>
    get<StudioProfileResponse>('/studios/me'),

  updateMyProfile: (data: StudioProfileUpdate) =>
    put<StudioProfileResponse>('/studios/me', data),

  getProfile: (id: string) =>
    get<StudioProfileResponse>(`/studios/${id}`),
};

// --- Job Posts -------------------------------------------

export const jobPosts = {
  create: (data: JobPostCreate) =>
    post<JobPostResponse>('/job-posts', data),

  list: (params?: Record<string, string>) => {
    const query = params ? '?' + new URLSearchParams(params).toString() : '';
    return get<JobPostListResponse>(`/job-posts${query}`);
  },

  listMine: () =>
    get<JobPostListResponse>('/job-posts/mine'),

  listWithMatching: (params?: Record<string, string> & {
    user_latitude?: string;
    user_longitude?: string;
    max_distance_km?: string;
  }) => {
    const query = params ? '?' + new URLSearchParams(params).toString() : '';
    return get<JobPostWithMatchingListResponse>(`/job-posts/for-me/with-matching${query}`);
  },

  get: (id: string) =>
    get<JobPostResponse>(`/job-posts/${id}`),

  update: (id: string, data: Partial<JobPostCreate>) =>
    put<JobPostResponse>(`/job-posts/${id}`, data),

  delete: (id: string) =>
    del<void>(`/job-posts/${id}`),
};

// --- Applications ----------------------------------------

export const applications = {
  apply: (jobPostId: string, data?: ApplicationCreate) =>
    post<ApplicationResponse>(`/job-posts/${jobPostId}/applications`, data || {}),

  getMyApplications: () =>
    get<{ items: ApplicationWithJobResponse[]; total: number }>('/applications/me'),

  withdraw: (id: string) =>
    post<ApplicationResponse>(`/applications/${id}/withdraw`),

  getForJobPost: (jobPostId: string) =>
    get<{ items: ApplicationWithInstructorResponse[]; total: number }>(`/job-posts/${jobPostId}/applications`),

  /** Accept application and reveal contact info (PMF pivot: replaces offer flow for urgent) */
  accept: (applicationId: string) =>
    post<ContactRevealResponse>(`/applications/${applicationId}/accept`),
};

// --- Offers ----------------------------------------------

export const offers = {
  create: (data: OfferCreate) =>
    post<OfferResponse>('/offers', data),

  getMyOffers: () =>
    get<OfferListResponse>('/offers/me'),

  accept: (id: string) =>
    post<OfferResponse>(`/offers/${id}/accept`),

  reject: (id: string) =>
    post<OfferResponse>(`/offers/${id}/reject`),
};

// --- Contracts -------------------------------------------

export const contracts = {
  createFromOffer: (offerId: string) =>
    post<ContractResponse>(`/contracts/from-offer/${offerId}`),

  getMyContracts: () =>
    get<ContractListResponse>('/contracts/me'),

  setInProgress: (id: string) =>
    post<ContractResponse>(`/contracts/${id}/set-in-progress`),

  confirmCompletion: (id: string) =>
    post<ContractResponse>(`/contracts/${id}/confirm-completion`),

  cancel: (id: string, data: ContractCancelRequest) =>
    post<ContractResponse>(`/contracts/${id}/cancel`, data),

  reportNoShow: (id: string, reportedUserId: string) =>
    post<ContractResponse>(`/contracts/${id}/report-no-show`, { reported_user_id: reportedUserId }),
};

// --- Reviews ---------------------------------------------

export const reviews = {
  create: (applicationId: string, data: ReviewCreate) =>
    post<ReviewResponse>(`/applications/${applicationId}/reviews`, data),

  getEligibility: (applicationId: string) =>
    get<ReviewEligibility>(`/applications/${applicationId}/reviews`),

  getMyReviewForApplication: (applicationId: string) =>
    get<ReviewResponse>(`/applications/${applicationId}/reviews/my`),

  update: (id: string, data: ReviewCreate) =>
    put<ReviewResponse>(`/reviews/${id}`, data),

  delete: (id: string) =>
    del<void>(`/reviews/${id}`),

  getReceived: () =>
    get<ReviewListResponse>('/reviews/received'),

  getWritten: () =>
    get<ReviewListResponse>('/reviews/written'),
};

// --- Subscriptions ---------------------------------------

export const subscriptions = {
  getStatus: () =>
    get<SubscriptionStatusResponse>('/subscriptions/me'),

  initializeUpgrade: () =>
    post<UpgradeInitializeResponse>('/subscriptions/upgrade'),

  confirmPayment: (data: { payment_key: string; order_id: string }) =>
    post<SubscriptionStatusResponse>('/subscriptions/confirm', data),

  cancel: (reason?: string) =>
    post<SubscriptionStatusResponse>('/subscriptions/cancel', reason ? { reason } : {}),

  getHistory: () =>
    get<unknown[]>('/subscriptions/history'),

  registerBillingKey: (data: { auth_key: string; customer_key: string }) =>
    post<BillingKeyRegisterResponse>('/subscriptions/billing/register', data),

  getBillingMethod: () =>
    get<BillingMethodResponse>('/subscriptions/billing'),

  removeBillingKey: () =>
    del<void>('/subscriptions/billing'),

  initBankTransfer: (data: { depositor_name: string }) =>
    post<BankTransferUpgradeResponse>('/subscriptions/upgrade/bank-transfer', data),
};

// --- Trust Tier ------------------------------------------

export const tier = {
  getMyTier: () =>
    get<TierResponse>('/tier/me'),

  getUserTier: (userId: string) =>
    get<TierPublicResponse>(`/tier/user/${userId}`),

  getRequirements: () =>
    get<TierRequirementsResponse>('/tier/requirements'),
};

// --- Penalties -------------------------------------------

export const penalties = {
  report: (data: PenaltyReportRequest) =>
    post<PenaltyRecordResponse>('/penalties/report', data),

  getMyPenalties: () =>
    get<{ items: PenaltyRecordResponse[]; total: number }>('/penalties/me'),
};

// --- Payment Confirmations -------------------------------

export const paymentConfirmations = {
  markPaid: (applicationId: string, data: MarkPaidRequest) =>
    post<PaymentConfirmationResponse>(`/applications/${applicationId}/mark-paid`, data),

  confirm: (id: string) =>
    post<PaymentConfirmationResponse>(`/payment-confirmations/${id}/confirm`),

  dispute: (id: string, reason: string) =>
    post<PaymentConfirmationResponse>(`/payment-confirmations/${id}/dispute`, { reason }),

  getMyConfirmations: () =>
    get<{ items: PaymentConfirmationResponse[] }>('/payment-confirmations/me'),
};

// --- Verification ----------------------------------------

export const verification = {
  requestPhone: (phone: string) =>
    post<{ message: string }>('/verification/phone/request', { phone }),

  verifyPhone: (phone: string, otp: string) =>
    post<{ message: string }>('/verification/phone/verify', { phone, otp }),

  verifyBusiness: (businessNumber: string) =>
    post<{ message: string }>('/verification/business/verify', { business_number: businessNumber }),

  getStatus: () =>
    get<VerificationStatusResponse>('/verification/status'),
};

// --- Profile Completeness --------------------------------

export const profileCompleteness = {
  get: () =>
    get<ProfileCompletenessResponse>('/profile/completeness'),

  checkAction: (action: string) =>
    get<{ allowed: boolean; reason?: string }>(`/profile/completeness/check/${action}`),

  uploadCertification: (file: File) => {
    const formData = new FormData();
    formData.append('file', file);
    return request<CertUploadResponse>('/profile/certifications', {
      method: 'POST',
      body: formData,
    });
  },
};

// --- Chat ------------------------------------------------
// PMF pivot: Chat disabled -- contact reveal replaces in-app chat
// export const chat = {
//   getThreads: () =>
//     get<ThreadResponse[]>('/threads'),
//
//   createThread: (data: ThreadCreate) =>
//     post<ThreadResponse>('/threads', data),
//
//   getMessages: (threadId: string) =>
//     get<MessageResponse[]>(`/threads/${threadId}/messages`),
//
//   sendMessage: (threadId: string, content: string) =>
//     post<MessageResponse>(`/threads/${threadId}/messages`, { content }),
// };

// --- Support ---------------------------------------------

export const support = {
  createTicket: (data: SupportTicketCreate) =>
    post<SupportTicketResponse>('/support/tickets', data),

  getMyTickets: () =>
    get<SupportTicketResponse[]>('/support/tickets/me'),
};

// --- Application Templates -------------------------------
// PMF pivot: Application templates disabled -- simplifying application flow
// export const applicationTemplates = {
//   list: () =>
//     get<ApplicationTemplateResponse[]>('/application-templates'),
//
//   create: (data: { name: string; content: string; is_default?: boolean }) =>
//     post<ApplicationTemplateResponse>('/application-templates', data),
//
//   update: (id: string, data: { name?: string; content?: string; is_default?: boolean }) =>
//     put<ApplicationTemplateResponse>(`/application-templates/${id}`, data),
//
//   delete: (id: string) =>
//     del<void>(`/application-templates/${id}`),
//
//   use: (id: string) =>
//     post<ApplicationTemplateResponse>(`/application-templates/${id}/use`),
//
//   getSuggestions: (jobType?: string) => {
//     const query = jobType ? `?job_type=${jobType}` : '';
//     return get<ApplicationTemplateResponse[]>(`/application-templates/suggestions${query}`);
//   },
// };

// --- Daily Usage -----------------------------------------
// Removed with Trust Tier pivot

// --- Handoff Notes ---------------------------------------

export const handoffNotes = {
  upsert: (jobPostId: string, data: HandoffNoteCreate) =>
    put<HandoffNoteFullResponse>(`/job-posts/${jobPostId}/handoff-note`, data),

  get: (jobPostId: string) =>
    get<HandoffNotePublicResponse | HandoffNoteFullResponse>(`/job-posts/${jobPostId}/handoff-note`),

  delete: (jobPostId: string) =>
    del<void>(`/job-posts/${jobPostId}/handoff-note`),
};

// --- Backup Instructors ----------------------------------

export const backupInstructors = {
  list: () =>
    get<{ items: BackupInstructorResponse[]; total: number }>('/studios/me/backup-instructors'),

  add: (data: BackupInstructorCreate) =>
    post<BackupInstructorResponse>('/studios/me/backup-instructors', data),

  update: (instructorId: string, data: BackupInstructorUpdate) =>
    patch<BackupInstructorResponse>(`/studios/me/backup-instructors/${instructorId}`, data),

  remove: (instructorId: string) =>
    del<void>(`/studios/me/backup-instructors/${instructorId}`),
};

// --- Convenience: grouped API ----------------------------

export const api = {
  auth,
  instructors,
  studios,
  jobPosts,
  applications,
  offers,
  contracts,
  reviews,
  subscriptions,
  tier,
  penalties,
  paymentConfirmations,
  verification,
  profileCompleteness,
  // PMF pivot: chat disabled -- contact reveal replaces in-app chat
  // chat,
  support,
  // PMF pivot: applicationTemplates disabled
  // applicationTemplates,
  // Daily usage removed with Trust Tier pivot
  // usage,
  handoffNotes,
  backupInstructors,
} as const;

export default api;
